include/nbl/ext/DebugDraw/CDrawAABB.h

```diff
@@ -0,0 +1,86 @@
+
+        //! optional, default MDI buffer allocated if not provided
+        core::smart_refctd_ptr<streaming_buffer_t> streamingBuffer = nullptr;
+    };
+    
+    struct SCreationParameters : SCachedCreationParameters
+    {
+        core::smart_refctd_ptr<asset::IAssetManager> assetManager = nullptr;
+
+        core::smart_refctd_ptr<video::IGPUPipelineLayout> pipelineLayout;
+        core::smart_refctd_ptr<video::IGPURenderpass> renderpass = nullptr;
+    };
+
+    // creates an instance that can draw one AABB via push constant or multiple using streaming buffer
+    static core::smart_refctd_ptr<DrawAABB> create(SCreationParameters&& params);
+
+    // creates default pipeline layout for push constant version
+    static core::smart_refctd_ptr<video::IGPUPipelineLayout> createDefaultPipelineLayout(video::ILogicalDevice* device, const asset::SPushConstantRange& pcRange);
+
+    // creates default pipeline layout for streaming version
+    static core::smart_refctd_ptr<video::IGPUPipelineLayout> createDefaultPipelineLayout(video::ILogicalDevice* device);
+
+    static core::smart_refctd_ptr<video::IGPUGraphicsPipeline> createDefaultPipeline(video::ILogicalDevice* device, video::IGPUPipelineLayout* layout, video::IGPURenderpass* renderpass, video::IGPUGraphicsPipeline::SShaderSpecInfo& vertex, video::IGPUGraphicsPipeline::SShaderSpecInfo& fragment);
+
+    //! mounts the extension's archive to given system - useful if you want to create your own shaders with common header included
+    static const core::smart_refctd_ptr<system::IFileArchive> mount(core::smart_refctd_ptr<system::ILogger> logger, system::ISystem* system, const std::string_view archiveAlias = "");
+
+    inline const SCachedCreationParameters& getCreationParameters() const { return m_cachedCreationParams; }
+
+    // records draw command for single AABB, user has to set pipeline outside
+    bool renderSingle(video::IGPUCommandBuffer* commandBuffer);
+
+    bool render(video::IGPUCommandBuffer* commandBuffer, video::ISemaphore::SWaitInfo waitInfo, float* cameraMat3x4);
+
+    static std::array<hlsl::float32_t3, 24> getVerticesFromAABB(const core::aabbox3d<float>& aabb);
+
+    void addAABB(const core::aabbox3d<float>& aabb, const hlsl::float32_t4& color = { 1,0,0,1 });
+    void addAABB(const hlsl::shapes::AABB<3,float>& aabb, const hlsl::float32_t4& color = { 1,0,0,1 });
+
+    void addOBB(const hlsl::shapes::AABB<3, float>& aabb, const hlsl::float32_t3x4 transform, const hlsl::float32_t4& color = { 1,0,0,1 });
+
+    void clearAABBs();
+
+protected:
+	DrawAABB(SCreationParameters&& _params, core::smart_refctd_ptr<video::IGPUGraphicsPipeline> pipeline);
+	~DrawAABB() override;
+
+private:
+    static core::smart_refctd_ptr<video::IGPUGraphicsPipeline> createPipeline(SCreationParameters& params);
+    static bool createStreamingBuffer(SCreationParameters& params);
+
+    std::vector<debugdraw::InstanceData> m_instances;
```

Comment:
immediate mode for GUI is ok, but I'd like to avoid it here, get rid of `add` and make `render` take a `span<const InstanceData>`